tests/apt/test_protocol.py

```diff
@@ -215,6 +218,63 @@ def test_AptMessage_MGMSG_MOD_IDENTIFY_to_bytes() -> None:
     assert msg.to_bytes() == b"\x23\x02\x01\x00\x50\x01"
 
 
+def test_AptMessage_MGMSG_MOT_GET_POSCOUNTER_from_bytes() -> None:
+    msg = AptMessage_MGMSG_MOT_GET_POSCOUNTER.from_bytes(
+        bytes.fromhex("1204 0600 A2 01 0100400D0300")
+    )
+
+    assert msg.destination == 0x22
```

Comment:
I see that there's an error in the official documentation where they say that `0xA2` should be channel 2 (`0x02`) after setting the highest bit to 0. Channel 2 would actually be transmitted as `0x82`.

It's fine to leave it as `0x22`, but it might be nice to leave a comment here about the error in the documentation.

Reply:
To be honest, I am not too sure what you mean by this. To explain myself, the reason I assert `msg.destination == 0x22` is because I have `destination=Address(destination & 0x7F)` in the original function. And the reason I `& 0x'F` is because the documentation looked similar to `AptMessage_MGMSG_HW_GET_INFO` from which I based most of my code off of. 

I'll post your explanation in a comment for now I guess. I hope that's fine.

Reply:
Your confusion is justified, as I was also confused. This was my mistake, although the documentation is also a problem.

On [page 63 of the protocol specification](https://www.thorlabs.com/Software/Motion%20Control/APT_Communications_Protocol.pdf) the destination in the example is called "Channel 2".

This is wrong, but not in the way that I thought it was wrong. I got confused.

It's wrong because, in the APT protocol, this is called an _address_, not a _channel_. An address describes what device is sending or receiving a message (source and destination). A channel describes which motor you are talking to on the device.

The documentation used the wrong term. It should say "Bay 1," instead of "Channel 2." [Bay 1's address is represented by `0x22`](https://github.com/moonshot-nagayama-pj/PnPQ/blob/main/src/pnpq/apt/protocol.py#L68). [Channel 2 is represented by `0x02`](https://github.com/moonshot-nagayama-pj/PnPQ/blob/main/src/pnpq/apt/protocol.py#L115).

I assume that the author of was counting from 1, not from 0, so what is called "Bay 1" in earlier parts of the specification became "Channel 2" here.

This is why it's important to make documentation precise and use vocabulary clearly and consistently.

As for why you are `& 0x7F`-ing the input value, read pages 34, 35, and 36 of the protocol specification.

Reply:
Yes. I understand now. But how come you are `& 0x7F`-ing instead of `| 0x80`-ing them? Wouldn't it have the same effect? Or, since the value is not strictly a 8bit integer, we are making sure that only the last seven least-significant bits are used. That makes sense.

Okay. I think I get it. I will merge now. Thank you.

Reply:
We do `| 0x80` the value before writing it out to bytes: https://github.com/moonshot-nagayama-pj/PnPQ/blob/main/src/pnpq/apt/protocol.py#L346

When reading in from bytes, we `& 0x7F` the value.

Reply:
Ohhhhh. That makes a lot of sense. Since `bin(0x80) = '0b10000000'` and `bin(0x7F) '0b01111111'`, we set the most significant bit when we write it, and then we remove the most significant bit when we read it. That makes a lot of sense. Okay. Thank you.